src/components/NavigationOutbox.vue

```diff
@@ -0,0 +1,66 @@
+<!--
+  - @copyright Copyright (c) 2022 Richard Steinmetz <richard@steinmetz.cloud>
+  -
+  - @author Richard Steinmetz <richard@steinmetz.cloud>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<AppNavigationItem
+		id="navigation-outbox"
+		key="navigation-outbox"
+		icon="icon-mail"
+		:title="t('mail', 'Outbox')"
+		:to="to">
+		<template #counter>
+			<CounterBubble v-if="count">
+				{{ count }}
+			</CounterBubble>
+		</template>
+	</AppNavigationItem>
+</template>
+
+<script>
+import AppNavigationItem from '@nextcloud/vue/dist/Components/AppNavigationItem'
+import CounterBubble from '@nextcloud/vue/dist/Components/CounterBubble'
+
+export default {
+	name: 'NavigationOutbox',
+	components: {
+		AppNavigationItem,
+		CounterBubble,
+	},
+	computed: {
+		to() {
+			return {
+				name: 'outbox',
+			}
+		},
+		count() {
+			// TODO: fix outbox initial state that doesnt happen when the page loads.
+			return Object.keys(this.$store.getters['outbox/getAllMessages']).length
+		},
```

Comment:
```suggestion
		count() {
			// TODO: fix outbox initial state that doesnt happen when the page loads.
			return this.$store.getters['outbox/getAllMessages'].length
		},
```
This should work too. The getter is already returning an array.

---

> - [ ] Fix counter initial state

FYI, we need to load the outbox messages via initial state to make this work on page load. However, it would be better performance wise to only load the count initially.

Reply:
Both done.